core/server/master/src/main/java/alluxio/master/job/CopyJob.java

```diff
@@ -0,0 +1,745 @@
+        .add("Bandwidth", mBandwidth)
+        .add("UsePartialListing", mUsePartialListing)
+        .add("VerificationEnabled", mVerificationEnabled)
+        .add("RetryBlocks", mRetryBlocks)
+        .add("FailedFiles", mFailedFiles)
+        .add("StartTime", mStartTime)
+        .add("ProcessedFileCount", mProcessedFileCount)
+        .add("LoadedByteCount", mLoadedByteCount)
+        .add("TotalBlockCount", mTotalBlockCount)
+        .add("CurrentBlockCount", mCurrentBlockCount)
+        .add("TotalFailureCount", mTotalFailureCount)
+        .add("CurrentFailureCount", mCurrentFailureCount)
+        .add("State", mState)
+        .add("BatchSize", BATCH_SIZE)
+        .add("FailedReason", mFailedReason)
+        .add("FileIterator", mFileIterator)
+        .add("CurrentFile", mCurrentFile)
+        .add("BlockIterator", mBlockIterator)
+        .add("EndTime", mEndTime)
+        .toString();
+  }
+
+  @Override
+  public Journal.JournalEntry toJournalEntry() {
+    alluxio.proto.journal.Job.CopyJobEntry.Builder jobEntry = alluxio.proto.journal.Job.CopyJobEntry
+        .newBuilder()
+        .setSrc(mSrc)
+        .setDst(mDst)
+        .setState(JobState.toProto(mState))
+        .setPartialListing(mUsePartialListing)
+        .setVerify(mVerificationEnabled)
+        .setJobId(mJobId);
+    mUser.ifPresent(jobEntry::setUser);
+    mBandwidth.ifPresent(jobEntry::setBandwidth);
+    mEndTime.ifPresent(jobEntry::setEndTime);
+    return Journal.JournalEntry
+        .newBuilder()
+        .setCopyJob(jobEntry.build())
+        .build();
+  }
+
+  /**
+   * Get duration in seconds.
+   * @return job duration in seconds
+   */
+  @VisibleForTesting
+  public long getDurationInSec() {
+    return (mEndTime.orElse(System.currentTimeMillis()) - mStartTime) / 1000;
+  }
+
+  @Override
+  public boolean processResponse(CopyTask loadTask) {
+    try {
+      long totalBytes = loadTask.getBlocks().stream()
+          .map(Block::getLength)
+          .reduce(Long::sum)
+          .orElse(0L);
+      LoadResponse response = loadTask.getResponseFuture().get();
+      if (response.getStatus() != TaskStatus.SUCCESS) {
+        LOG.debug(format("Get failure from worker: %s", response.getBlockStatusList()));
+        for (BlockStatus status : response.getBlockStatusList()) {
+          totalBytes -= status.getBlock().getLength();
+          if (!isHealthy() || !status.getRetryable() || !addBlockToRetry(
+              status.getBlock())) {
+            addBlockFailure(status.getBlock(), status.getMessage(), status.getCode());
+          }
+        }
+      }
+      addLoadedBytes(totalBytes);
+      JOB_LOAD_BLOCK_COUNT.inc(
+          loadTask.getBlocks().size() - response.getBlockStatusCount());
+      JOB_LOAD_BLOCK_SIZE.inc(totalBytes);
+      JOB_LOAD_RATE.mark(totalBytes);
+      return response.getStatus() != TaskStatus.FAILURE;
+    }
+    catch (ExecutionException e) {
+      LOG.warn("exception when trying to get load response.", e.getCause());
```

Comment:
copy response

Reply:
please see PR description that copyjob is still in progress so ignore it.